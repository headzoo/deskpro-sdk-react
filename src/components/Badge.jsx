import React from 'react';
import PropTypes from 'prop-types';
import { CircleBadge } from 'deskpro-components';

const Badge = ({ count, visible, style, ...props }) => {
  if (!visible) {
    return null;
  }

  const styles = Object.assign({}, {
    position: 'absolute',
    top:      -13,
    right:    -10
  }, style);

  return (
    <div {...props} style={styles}>
      <CircleBadge type="danger" max={99}>
        {count}
      </CircleBadge>
    </div>
  );
};

Badge.propTypes = {
  /**
   * Whether or not the badge count is displayed.
   */
  visible: PropTypes.bool,

  /**
   * The badge count.
   */
  count: PropTypes.number,

  /**
   * Additional styles passed to the room element.
   */
  style: PropTypes.object
};

Badge.defaultProps = {
  visible: false,
  count:   0,
  style:   {}
};

export default Badge;
