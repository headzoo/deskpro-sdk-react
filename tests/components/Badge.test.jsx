import React from 'react';
import renderer from 'react-test-renderer';
import Badge from '../../src/components/Badge';

it('Badge is visible', () => {
  const renderedValue = renderer.create(
    <Badge count={3} visible />
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});

it('Badge is hidden', () => {
  const renderedValue = renderer.create(
    <Badge count={3} visible={false} />
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
