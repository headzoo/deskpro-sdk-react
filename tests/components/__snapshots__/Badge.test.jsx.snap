// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge is hidden 1`] = `null`;

exports[`Badge is visible 1`] = `
<div
  style={
    Object {
      "position": "absolute",
      "right": -10,
      "top": -13,
    }
  }
>
  <div
    className="dp-badge dp-badge--danger dp-bg--danger dp-circle-badge"
  >
    3
  </div>
</div>
`;
